src/methods/clipping/clipping_processor.jl

```diff
@@ -0,0 +1,286 @@
+        if prev_counter < counter
+            # Order intersection points based on alpha values
+            new_order = sortperm(alpha_a_list[prev_counter:counter-1])
+            pts_to_add = Array{PolyNode, 1}(undef, counter - prev_counter)
+            for kk in eachindex(new_order)
+                # Create PolyNodes of the new intersection points in the correct order
+                # and store the correct index in a_idx_list
+                pts_to_add[new_order[kk]] = PolyNode(prev_counter+kk-1, true, 0, false, alpha_a_list[prev_counter+kk-1])
+                if prev_counter+kk-1 <= length(a_idx_list)
+                    a_idx_list[prev_counter+kk-1] = acount + new_order[kk] - 1
+                else
+                    length_diff = prev_counter+kk-1-length(a_idx_list)
+                    append!(a_idx_list, zeros(length_diff))
+                    a_idx_list[prev_counter+kk-1] = acount + new_order[kk] - 1
+                end
+            end
+
+            # Add the PolyNodes to a_list and update acount
+            splice!(a_list, acount:(acount-1), pts_to_add)
+            acount = acount + counter - prev_counter
+            
+        end
+        
+    end
+
+    # Truncate the inter_related lists
+
+    intr_list = intr_list[1:counter-1] 
+    a_idx_list = a_idx_list[1:counter-1]
+    b_idx_list = b_idx_list[1:counter-1]
+    alpha_a_list = alpha_a_list[1:counter-1]
+    alpha_b_list = alpha_b_list[1:counter-1]
+
+    return intr_list, a_idx_list, b_idx_list, alpha_a_list, alpha_b_list
+end
+
+"""
+    _build_b_list(
+        b_list, intr_list, 
+        a_idx_list, b_idx_list, 
+        alpha_b_list
+    )::b_neighbors, b_list
+
+    This function builds b_list. Note that after calling this function, b_list
+    is not fully updated. The entry/exit flags still need to be updated. The variables poly_a and poly_b are the original polygons that the
+    user passed in. The variables edges_a and edges_b are lists of the edges of each polygon, 
+    in the form of their cartesian points. Then intr_list is a list of the intersection points
+    in their cartesion point form. Then a_idx_list is a list of the indicies of where in a_list
+    an intersection point lies. The value at index i of a_idx_list is the location in a_list where
+    the ith point of intr_list lies. Then b_idx_list indicates which EDGE of poylgon b an intersection
+    point lies. Then alpha_b_list is a veector of scalar values which help order
+    the intersection points in the case that they lie on the same edge of a polygon. Finally,
+    b_list starts out as an array of PolyNodes only containing the original points of poly_b
+    but after this function is run it include intersection points to.
+"""
+
+function _build_b_list(b_list, intr_list, a_idx_list, b_idx_list, alpha_b_list)
+    # Iterate through the b_list and add in intersection points
+    # Occasionally I need to skip the new points I added to the array
+    skip = false
```

Comment:
So this function should now take in `a_list`, `a_idx_list`, and `poly_b`. You will want to loop through the edges of `b` like you do above with `a`, but you will also need the index (`enumerate` or a counter). Then, check if any points in `a_idx_list` need to be added by checking if the neighbor field is equal to the edge index. You can then update `a_list` neighbor fields as you go to their index in b. When creating the `b_list` polyNodes, don't forget to swap the alpha/beta values. (Although do we even need them anymore? We could maybe get rid of that field if we only use it when making the `a_list`)